src/BSeries.jl

```diff
@@ -1485,8 +1485,9 @@ This code is based on the Theorem 2 of
   [DOI: 10.1007/s10208-010-9073-1](https://link.springer.com/article/10.1007/s10208-010-9073-1)
  """
 function is_energy_preserving(map_series)
+    #check if the first element is map or flow
+    @assert typeof(map_series) == TruncatedBSeries{RootedTree{Int64, Vector{Int64}}, Rational{Int64}}
```

Comment:
This just checks that the input is a BSeries.  What we want to check is whether the BSeries represents a map or a flow, based on whether the first coefficient is 1 or zero.  We discussed this last week in my office.